plugins/filter-highlight-all/prism-filter-highlight-all.js

```diff
@@ -0,0 +1,106 @@
+(function () {
+
+	if (typeof self !== 'undefined' && !self.Prism) {
+		return;
+	}
+
+	// https://developer.mozilla.org/en-US/docs/Web/API/Element/matches#Polyfill
+	if (!Element.prototype.matches) {
+		Element.prototype.matches = Element.prototype.msMatchesSelector || Element.prototype.webkitMatchesSelector;
+	}
+
+	var script = Prism.util.currentScript();
+
+
+	/**
+	 * @type {Array<(element: HTMLElement) => boolean>}
+	 */
+	var filters = [];
+
+	var config = Prism.plugins.filterHighlightAll = {
+
+		/**
+		 * Filters the elements of `highlightAll` and `highlightAllUnder` such that only element for which the given
+		 * function returns `true` will be highlighted.
+		 *
+		 * @param {(value: { element: HTMLElement, language: string }) => boolean} condition
+		 */
+		filter: function (condition) {
+			filters.push(function (element) {
+				return condition({
+					element: element,
+					language: Prism.util.getLanguage(element)
+				});
+			});
+		},
+
+		/**
+		 * Filters the elements of `highlightAll` and `highlightAllUnder` such that only element which match the given
+		 * CSS selection will be highlighted.
+		 *
+		 * @param {string} selector
+		 */
+		filterCss: function (selector) {
+			filters.push(function (element) {
+				return element.matches(selector);
+			});
+		},
+
+		/**
+		 * Filters the elements of `highlightAll` and `highlightAllUnder` such that only element which do not match the
+		 * given CSS selection will be highlighted.
+		 *
+		 * @param {string} selector
+		 */
+		excludeCss: function (selector) {
```

Comment:
Then this one could be `rejectSelector`? And/or make the above one `acceptSelector`? `filterSelector` would be ok, cuz that does match the semantics of the array method `filter`.

Reply:
I don't like the "accept"/"reject"-pair because it reminds me of network requests.
I want to keep the "filter" but you're right that "filter" and "exclude" aren't exactly opposites.

Or what about "matchesSelector" and "notMatchesSelector"? Quite verbose but clear.

The only thing now, that isn't clear from the name is that all filters and combined with a logical AND. But that should be in the doc.

Reply:
I was thinking about it as the array functions:

https://lodash.com/docs/4.17.15#filter
https://lodash.com/docs/4.17.15#reject

Hence `filterSelector` & `rejectSelector`.

`includeSelector` & `excludeSelector` could also work.

Reply:
`filterSelector` and `rejectSelector` sounds good to me.